src/persist/src/s3.rs

```diff
@@ -44,17 +45,16 @@ impl S3BlobConfig {
         prefix: String,
         role_arn: Option<String>,
     ) -> Result<Self, Error> {
-        let mut config = AwsConfig::load_from_env().await;
+        let mut loader = aws_config::from_env();
         if let Some(role_arn) = role_arn {
-            let provider = AssumeRoleProvider::builder(role_arn).session_name("persist");
-            let provider = if let Some(region) = config.region() {
```

Comment:
do you need to do anything for `region`? or is that fine because it's in the `loader` and we do `AwsConfig::from_loader`?

Reply:
Yeah I was just tracing through that. I think I actually need to install the default region chain :S.

Reply:
Ok, updated to handle that.